sql-parser/src/main/java/io/crate/sql/tree/SearchedCaseExpression.java

```diff
@@ -32,8 +29,7 @@
     private final Expression defaultValue;
 
     public SearchedCaseExpression(List<WhenClause> whenClauses, Expression defaultValue) {
-        Preconditions.checkNotNull(whenClauses, "whenClauses is null");
-        this.whenClauses = ImmutableList.copyOf(whenClauses);
```

Comment:
we still want to create an immutable copy of the list though no?

Reply:
i changed it to be unmodifiable now